nmostesting/ControllerTest.py

```diff
@@ -184,16 +182,20 @@ def _send_testing_facade_questions(
         answers:    list of all possible answers
         test_type:  "single_choice" - one and only one answer
                     "multi_choice" - multiple answers
-                    "action" - Test User asked to click button, defaults to self.question_timeout
-        timeout:    number of seconds before Testing Façade times out test
+                    "action" - Test User asked to click button
+        timeout:    number of seconds before Testing Façade times out test. A value of 0 disables timeout mechanism
         multipart_test: indicates test uses multiple questions. Default None, should be increasing
                     integers with each subsequent call within the same test
         metadata: Test details to assist fully automated testing
         """
 
         method = getattr(self, test_method_name)
 
-        question_timeout = timeout if timeout else self.question_timeout
+        question_timeout = timeout or CONFIG.CONTROLLER_TESTING_TIMEOUT
+
+        # If timeout == 0 then Testing Facade timeout mechanism is disabled
+        question_timeout = question_timeout if question_timeout > 0 else None
```

Comment:
I'm still a bit confused whether `None` and `0` are treated the same or not and what the default behaviour of this function is supposed to be. The default value for `timeout` in the function signature is still `None`, and the description now says `0` disables timeout mechanism.

Does that mean timeout is disabled by default?

No, and the new description is also misleading I think... because this line just checks `timeout` truthiness:

```
        question_timeout = timeout or CONFIG.CONTROLLER_TESTING_TIMEOUT
```

(Python truthiness also means that in the second line `question_timeout if question_timeout > 0 else None` could be written as `question_timeout or None`.)

Does this function need both a mechanism to request the CONTROLLER_TESTING_TIMEOUT default AND a mechanism to request no timeout, or is that not necessary?